lib/delivery_boy/instance.rb

```diff
@@ -115,5 +116,14 @@ def producer_options
         transactional_timeout: config.transactional_timeout,
       }
     end
+
+    def sasl_oauth_token_provider
+      return if config.sasl_oauth_token_provider.nil?
+
+      config.sasl_oauth_token_provider
+        .split('::')
+        .reduce(Module, :const_get)
+        .new(config.client_id, config.secret, config.sasl_oauth_token_url)
```

Comment:
I’m not quite sure how this is supposed to work. Can you add a test for this logic?

Reply:
sure. I will make the necessary changes to make it clearer.